Assets/Mirror/Components/NetworkTransformBase.cs

```diff
@@ -174,267 +308,157 @@ void DeserializeFromReader(NetworkReader reader)
         }
 
-        // local authority client sends sync message to server for broadcasting
-        [Command]
-        void CmdClientToServerSync(byte[] payload)
+        // set position carefully depending on the target component
+        void ApplyPositionRotationScale(Vector3 position, Quaternion rotation, Vector3 scale)
         {
-            // Ignore messages from client if not in client authority mode
-            if (!clientAuthority)
-                return;
-
-            // deserialize payload
-            using (PooledNetworkReader networkReader = NetworkReaderPool.GetReader(payload))
-                DeserializeFromReader(networkReader);
-
-            // server-only mode does no interpolation to save computations,
-            // but let's set the position directly
-            if (isServer && !isClient)
-                ApplyPositionRotationScale(goal.localPosition, goal.localRotation, goal.localScale);
-
-            // set dirty so that OnSerialize broadcasts it
-            SetDirtyBit(1UL);
+            // local position/rotation for VR support
+            if (applyPosition) targetTransform.localPosition = position;
+            if (applyRotation) targetTransform.localRotation = rotation;
+            if (applyScale) targetTransform.localScale = scale;
         }
 
         // where are we in the timeline between start and goal? [0,1]
-        static float CurrentInterpolationFactor(DataPoint start, DataPoint goal)
+        Vector3 InterpolatePosition(DataPoint start, DataPoint goal, Vector3 currentPosition)
         {
-            if (start != null)
-            {
-                float difference = goal.timeStamp - start.timeStamp;
-
-                // the moment we get 'goal', 'start' is supposed to
-                // start, so elapsed time is based on:
-                float elapsed = Time.time - goal.timeStamp;
-                // avoid NaN
-                return difference > 0 ? elapsed / difference : 0;
-            }
-            return 0;
-        }
+            if (!interpolatePosition)
+                return currentPosition;
 
-        static Vector3 InterpolatePosition(DataPoint start, DataPoint goal, Vector3 currentPosition)
-        {
-            if (start != null)
+            if (start.movementSpeed != 0)
             {
-                // Option 1: simply interpolate based on time. but stutter
-                // will happen, it's not that smooth. especially noticeable if
-                // the camera automatically follows the player
-                //   float t = CurrentInterpolationFactor();
-                //   return Vector3.Lerp(start.position, goal.position, t);
+                // Option 1: simply interpolate based on time, but stutter will happen, it's not that smooth.
+                // This is especially noticeable if the camera automatically follows the player
+                // -         Tell SonarCloud this isn't really commented code but actual comments and to stfu about it
```

Comment:
sonarcloud is meant to help us, not make our comments more complicated :)

Reply:
Agreed.